src/json/from_json.cr

```diff
@@ -255,7 +255,7 @@ end
 # Raises `JSON::ParseException` if the deserialization fails.
 def Enum.new(pull : JSON::PullParser)
   {% if @type.annotation(Flags) %}
-    value = {{ @type }}::None
+    value = {{ @type }}.new(0)
```

Comment:
Consider not changing it to zero. If someone redefined `None` to not be zero, would they really want JSON parsing to create a zero?

Reply:
Oh, I see that it needs to start as empty in order to be able to be populated. So maybe nevermind, or maybe we should indeed still replace this with `None` if there are no flags.

Reply:
Exactly. This initializes an empty flag enum. An empty flag enum *always* has the value `0`. This is an implicit property of flag enums. We can use that here directly. "none" is just a name we assign to that.

But that name can collide with a member that's supposed to have that name. The motivation for this PR is for member name `None` to represent different values than `0`.

If we want to abstract the implicit property, we could consider adding a class method for the empty flag enum, similar to `.all_flags` (maybe `.no_flags` or `.empty`?).